classes/log/store.php

```diff
@@ -82,7 +127,25 @@ public function get_max_batch_size() {
         return $this->get_config('maxbatchsize', 100);
     }
 
+    /**
+     * Take rows from logstore_standard_log for the emit_task or failed_task
+     * and add in the logstore_standard_log_id and set the type.
+     *
+     * Return the persistent logstore_standard_log id from the database.
+     * @param array $events raw event data
+     * @return array
+     */
+    private function get_persistent_eventids(array $events) {
+        foreach ($events as $event) {
+            $eventid = $this->get_event_id($event);
```

Comment:
what happens if `get_event_id` returns 0?

Reply:
We insert the value 0 into the database. 0 means row not found. Prevents any problems with nulls.